components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+        }
+
+        /**
+         * Gets schema
+         * @returns {object} schema
+         */
+        get schema() {
+            return this._schema;
+        }
+
+        /**
+         * Sets schema
+         * @param {object} value, schema
+         */
+        set schema(value) {
+            this._schema = value;
+        }
+
+        /**
+         * Gets header value
+         * @returns {string} header
+         */
+        get headers() {
+            return (this.getAttribute('headers') || '').split(',');
+        }
+
+        /**
+         * Sets headers
+         * @param {array} headers to set
+         */
+        set headers(headers) {
+            // eslint-disable-next-line no-param-reassign
+            headers = Array.isArray(headers) ? headers.join(',') : headers;
+
+            if (headers === '') {
+                this.removeAttribute('headers');
+            } else {
+                this.setAttribute('headers', headers);
+            }
+        }
+
+        /**
+         * Gets first row
+         * @returns {string} firstrow seperated by pipe
+         */
+        get firstRow() {
+            return (this.getAttribute('first-row') || '').split('|');
+        }
+
+        /**
+         * sets rawFile value
+         * @param {array} values for firstrow
+         */
+        set firstRow(values) {
+            // eslint-disable-next-line no-param-reassign
+            values = Array.isArray(values) ? values.join('|') : values;
+
+            if (values === '') {
+                this.removeAttribute('first-row');
+            } else {
+                this.setAttribute('first-row', values);
+            }
+        }
+
+
+        /**
+         * Gets hasData attribute
+         * @returns {string} hasData attribute
+         */
+        get hasData() {
+            return this.getAttribute('has-data');
+        }
+
+        /**
+         * Sets hasData attribute
+         * @param {boolean} value, boolean
+         */
+        set hasData(value) {
+            this.setAttribute('has-data', value === true);
+        }
+
+        /**
+         * Gets hasErrors attribute
+         * @param {string} hasErrors attribute
+         */
+        get hasErrors() {
+            return this.getAttribute('has-errors');
+        }
+
+        /**
+         * Sets hasErrors attribute
+         * @param {boolean} value, boolean
+         */
+        set hasErrors(value) {
+            this.setAttribute('has-errors', value === true);
+        }
+
+        /**
+         * Gets import type
+         * @param {string} type of import
+         */
+        get type() {
+            return this.getAttribute('type') || 'csv';
+        }
+
+        /**
+         * Sets type attribute
+         * @param {string} value - type of import
+         */
+        set type(value) {
+            // ensure its always csv or xlsx
+            // eslint-disable-next-line no-param-reassign
+            value = (value !== 'xlsx') ? value = 'csv' : 'xlsx';
```

Comment:
wow... interesting problem actually - invalid values as props and attributes on elements. we ought to try to be consistent with in-built elements i guess.

anyway, let's introduce a new const here to avoid the eslint-disable (yes, we're not refactoring but we are trying to make it conform with linting rules not just copy the same code and add disables everywhere).